cirq/study/sweeps_test.py

```diff
@@ -74,6 +74,48 @@ def test_product():
     assert _values(sweep, 'b') == [4, 5, 6, 7, 4, 5, 6, 7, 4, 5, 6, 7]
 
 
+def test_slice_access_error():
+    sweep = cirq.Points('a', [1, 2, 3])
+    with pytest.raises(TypeError, match='<class \'str\'>'):
+        _ = sweep['junk']
+
+
+def test_slice_sweep():
+    sweep = cirq.Points('a', [1, 2, 3]) * cirq.Points('b', [4, 5, 6, 7])
+
+    first_two = sweep[:2]
+    assert list(first_two.param_tuples())[0] == (('a', 1), ('b', 4))
+    assert list(first_two.param_tuples())[1] == (('a', 1), ('b', 5))
+    assert len(list(first_two)) == 2
```

Comment:
Maybe add a test case for a slice of length 1 (should give sweep of len 1, not param resolver).

Reply:
added.